### ISSUE TITLE
Failed to restore plugin

### ISSUE THREAD

**Issue by netoolii** [moust/cordova-plugin-videoplayer#33]
When i try install this plugin, this error appear:

Failed to restore plugin "com.moust.cordova.videoplayer" from config.xml. You might need to try adding it again. Error: Failed to fetch plugin com.moust.cordova.videoplayer@https://github.com/moust/cordova-plugin-videoplayer via registry.
Probably this is either a connection problem, or plugin spec is incorrect.
Check your connection and plugin name/version/URL.
Failed to get absolute path to installed module

Can you help me ?

info:

node : v6.10.3
cordova: 7.0.1
ionic: 3.2.0
OS: MACOSX

**Comment by shopen21** [comment]
+1
Experience the same issue

**Comment by henryguyu** [comment]
Same issue

**Comment by moust** [comment]
I published the plugin on NPM repository to be able to install it again with Cordova cli.
However I was forced to change the name of the plugin due to an other existing project on NPM with the same name.
So you can now install it again with `cordova plugin add cordova-plugin-video-player` command.

**Comment by jmtech-online** [comment]
Just tried... defs not a connection error.

cordova plugin add cordova-plugin-video-player
codeError: Failed to fetch plugin git+https://github.com/moust/cordova-plugin-videoplayer.git via registry.
Probably this is either a connection problem, or plugin spec is incorrect.
Check your connection and plugin name/version/URL.
Failed to get absolute path to installed module

Node:  v6.10.3
Cordova: 7.0.1
Ionic: 2.2.3
OS: Mac Sierra

**Comment by moust** [comment]
Did you try to remove plugin from node_modules directory and execute `npm cache clear` before reinstall it ?

**Comment by ZalakainTheOne** [comment]
Thanks @moust that works.

**Comment by carponte** [comment]
I removed the plugin from node_modules directory, executed npm cache clear and tried to reinstall it.

I get: 

"Error: Failed to fetch plugin git+https://github.com/moust/cordova-plugin-videoplayer.git via registry.
Probably this is either a connection problem, or plugin spec is incorrect.
Check your connection and plugin name/version/URL.
Failed to get absolute path to installed module"

Any suggestion?

**Comment by francescodoffizi** [comment]
Same issue here

**Comment by tfairfield** [comment]
Same here. I am publishing an ionic app. For Android and iOS there is no issue, but publishing to Windows gets the same error as reported here:

`Failed to restore plugin "com.moust.cordova.videoplayer" from config.xml. You mi
ght need to try adding it again. Error: Failed to fetch plugin com.moust.cordova
.videoplayer@https://github.com/moust/cordova-plugin-videoplayer.git via registr
y.
Probably this is either a connection problem, or plugin spec is incorrect.
Check your connection and plugin name/version/URL.
Error: cmd: Command failed with exit code 1 Error output:
npm ERR! git clone C:\Users\IEUser\AppData\Roaming\npm-cache\_git-remotes\git-ht
tps-github-com-moust-cordova-plugin-videoplayer-git-5f632a25 C:\Users\IEUser\App
Data\Local\Temp\npm-548-0cfdf0c0\git-cache-bfaa6676\4e97e063ecddde4c3814605011fc
3f0a9d72a35e: Cloning into 'C:\Users\IEUser\AppData\Local\Temp\npm-548-0cfdf0c0\
git-cache-bfaa6676\4e97e063ecddde4c3814605011fc3f0a9d72a35e'`

My info:
cli packages: (C:\Users\IEUser\AppData\Roaming\nvm\v8.5.0\node_modules)
    @ionic/cli-utils  : 1.11.0
    ionic (Ionic CLI) : 3.11.0

global packages:
    cordova (Cordova CLI) : 7.0.1

local packages:
    @ionic/app-scripts : 1.3.12
    Cordova Platforms  : windows 5.0.0
    Ionic Framework    : ionic-angular 3.5.0

System:
    Node : v8.5.0
    npm  : 4.6.1
    OS   : Windows 8.1

**Comment by alexreg** [comment]
I have experienced this issue with other Cordova plugins on Cordova 7.1.0... so it seems to be a general bug with Cordova.

**Comment by afdev82** [comment]
any solution?

**Comment by tusharnetset** [comment]
Any solution

**Comment by PhilAndrew** [comment]
Use cordova version 8 and the problem goes away

**Comment by afdev82** [comment]
That's not true, I'm using cordova 8.0.0 and as long as the plugin is already in the node_modules folder I get the error.
I'm using also other npm packages in my project, so the solution that I came to is to execute the cordova command before `npm install` that I use to install also the other packages.
The reason is that `npm install` will install also the cordova plugins in the `node_modules` directory and after that the cordova command will fail.

**Comment by nickhingston** [comment]
seems to be to do with git adding "git+" to the repo specs.

if you remove that from both your config.xml and package.json - it works...(but they are then re-added for next time!)

Why?!
